src/FluentCheck.ts

```diff
@@ -13,11 +13,20 @@ class FluentResult {
   }
 }
 
-export type FluentConfig = { sampleSize?: number, shrinkSize?: number }
+export type FluentConfig = { sampleSize?: number, shrinkSize?: number}
 
 export class FluentCheck<Rec extends ParentRec, ParentRec extends {}> {
-  constructor(protected readonly parent: FluentCheck<ParentRec, any> | undefined = undefined,
+  public prng: {
+    generator: () => number,
+    seed: number | undefined
+  }
+
+  constructor(readonly parent: FluentCheck<ParentRec, any> | undefined = undefined,
     public readonly configuration: FluentConfig = {sampleSize: 1000, shrinkSize: 500}) {
+    if (this.parent === undefined)
```

Comment:
Use null coalescing to simplify this:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing

Reply:
I didn't know this existed but it's interesting and I will start using it in the future, however, I don't believe this a case where it can be used because this.parent is the one being checked if it is set and this.prng is the one being assigned. Maybe I understood incorrectly how nullish coalescing works?

Reply:
Not sure if this works:

`this.parent?.prng || { generator: Math.random, seed: undefined }`

Reply:
Also, take a look at this for future reference:

https://linguinecode.com/post/how-to-solve-typescript-possibly-undefined-value